### PR TITLE
[CI] Release artifacts job: binaries and containers

### PR DESCRIPTION
**Pull request by okdas**
## Summary

Provides binaries for releases and production-ready container images.

## Issue

- #495

## Type of change

Select one or more:

- [ ] New feature, functionality or library
- [ ] Bug fix
- [ ] Code health or cleanup
- [ ] Documentation
- [ ] Other (specify)

## Testing

**Documentation changes** (only if making doc changes)
- [ ] `make docusaurus_start`; only needed if you make doc changes

**Local Testing** (only if making code changes)
- [x] **Unit Tests**: `make go_develop_and_test`
- [x] **LocalNet E2E Tests**: `make test_e2e`
- See [quickstart guide](https://dev.poktroll.com/developer_guide/quickstart) for instructions

**PR Testing** (only if making code changes)
- [x] **DevNet E2E Tests**: Add the `devnet-test-e2e` label to the PR.
    - **THIS IS VERY EXPENSIVE**, so only do it after all the reviews are complete.
    - Optionally run `make trigger_ci` if you want to re-trigger tests without any code changes
    - If tests fail, try re-running failed tests only using the GitHub UI as shown [here](https://github.com/pokt-network/poktroll/assets/1892194/607984e9-0615-4569-9452-4c730190c1d2)

## Sanity Checklist

- [ ] I have tested my changes using the available tooling
- [ ] I have commented my code
- [ ] I have performed a self-review of my own code; both comments & source code
- [ ] I create and reference any new tickets, if applicable
- [ ] I have left TODOs throughout the codebase, if applicable

### CODE REVIEW

File: .github/workflows/release-artifacts.yml [hunk 1]
```diff
@@ -0,0 +1,86 @@
+name: Release artifacts
+
+on:
+  push:
+    tags:
+      - "v*.*.*"
```
Comment (okdas) [inline .github/workflows/release-artifacts.yml:6]: I've tested this on the `push` trigger, but we don't have a good way to test the release creation from tag. Let's create the first tag after this is merged to test this CI job fully.

**Comment by okdas** [comment]
@Olshansk Upgrade to 1.22 was important for integration with the `heighliner`. I put the integration on pause, but I kept the upgraded version since I don't see any reason not to — Golang has great backwards compatibility and all tests seem to pass fine.

File: Dockerfile.release [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+ARG TARGETARCH
+
+# Use `1025` G/UID so users can switch between this and `heighliner` image without a need to chown the files.
+RUN addgroup --gid 1025 -S pocket && adduser --uid 1025 -S pocket -G pocket
+
+COPY --chown=pocket:pocket release_binaries/poktroll_linux_$TARGETARCH /bin/poktrolld
+
+USER pocket
+
+ENTRYPOINT ["poktrolld"]
```
Comment (Olshansk) [inline Dockerfile.release:13]: Why is the poktrolld binary the entrpoint?

**Review by Olshansk — APPROVED** [review]
Just one non-blocking question!

File: Dockerfile.release (hunk 2 above)
Comment (okdas) [reply Dockerfile.release:13]: > Why is the poktrolld binary the entrpoint?

glad you asked. 

It allows to easy set up a shell alias to use `poktrolld` from docker without installing the binary on the machine. I'll showcase this in the docs.

Comment (Olshansk) [reply Dockerfile.release:13]: #PUC

```
# poktrolld is the entrypoint to allow easily setting up a shell alias to use poktrolld from docker without installing the binary on the machine. I'll showcase this in the docs.
ENTRYPOINT ["poktrolld"]
```

Comment (Olshansk) [reply Dockerfile.release:13]: @okdas ☝️

**Pull request merged**